### ISSUE TITLE
Errors in PIV module with "VOLUME" argument

### ISSUE THREAD

**Issue by fanpeizhou** [plumed/plumed2#883]
**Plumed version: 2.7.4**
**Description**: When adding "VOLUME" argument in PIV module, plumed will output: "_ERROR in input to action PIV with label piv0 : problem reading SWITCH1 keyword : R_0 is required_". But I do define the R_0 (see below), and if you delete the "VOLUME" line, it works well.
`UNITS LENGTH=A TIME=ps ENERGY=eV

PIV ...
LABEL=piv0
PRECISION=1000
VOLUME=32768
REF_FILE=fluid.pdb
PIVATOMS=1
ATOMTYPES=1
ONLYDIRECT
SWITCH1={RATIONAL R_0=6.0 MM=12 NN=4}
SORT=1
NLIST
NL_CUTOFF=10.0
NL_SKIN=1.0
NL_STRIDE=1
... `
 And my buddy also helped me to test a lot in the latest version of plumed. He found if you change the R_0 to 0.6 or 0.anything, the bug is gone. So I think this is, probably, a bug. Also might be our misoperation. 

Really appreciate it if anyone can help us out with this.

**Comment by GiovanniBussi** [comment]
I think there's a problem at these lines:
https://github.com/plumed/plumed2/blob/da2598c6de3f48dc48bdc60555892e1318386fb4/src/piv/PIV.cpp#L797-L807

Apparently, the input line is manipulated to replace `R_0` with something else.

I was checking this to verify there was not a bug in the general plumed parser. This seems specific for the PIV functionality. So I think @SilvioPipolo might have some better suggestion

**Comment by SilvioPipolo** [comment]
Dear Giovanni,
this is to scale R_0 by a proper volume-dependent factor when the same
scaling factor is used for scaling atom-atom distances in the PIV
definition. I think I had to use this solution (changing the input) for
restarts because this way R_0 is modified once and for all when the
simulation starts and then simply read from input if the simulation is
restarted.
I hope this helps,
best,
Silvio

**Comment by GiovanniBussi** [comment]
@SilvioPipolo I think the logic should be the opposite:
- first parse to `std::string`
- then convert to `double`
- then replace with the length of the original `std::string`.

The way to code is, it does not work if `R_0=6.0` because it prints it back as `R_0=6` and replace an incorrect part of the string.

I tried to fix this, see branch [fix-883](https://github.com/plumed/plumed2/tree/fix-883). I also modified a regtest to test this.

Let me know what you think.
